pkg/registry/blobs.go

```diff
@@ -46,11 +50,26 @@ func isBlob(req *http.Request) bool {
 
 // blobs
 type blobs struct {
-	// Blobs are content addresses. we store them globally underneath their sha and make no distinctions per image.
-	contents map[string][]byte
 	// Each upload gets a unique id that writes occur to until finalized.
 	uploads map[string][]byte
 	lock    sync.Mutex
+
+	bh BlobHandler
+}
+
+// BlobHandler is the interface for the storage layer underneath this registry.
+type BlobHandler interface {
```

Comment:
If `Store` can return an error, shouldn't `Stat` and `Get`?

Reply:
done